benches/generators.rs

```diff
@@ -102,3 +103,40 @@ fn init_jitter(b: &mut Bencher) {
         black_box(JitterRng::new().unwrap());
     });
 }
+
+
+
+#[bench]
+fn reseeding_xorshift_bytes(b: &mut Bencher) {
+    let mut rng = ReseedingRng::new(XorShiftRng::new().unwrap(),
```

Comment:
Wouldn't it make more sense to benchmark the PRNG we're interested in (ISAAC or HC128)? Especially given that your `BlockRng` idea integrates tighter with the PRNG algorithm.

Reply:
With HC-128 the overhead of reseeding is much larger:
```
test gen_bytes_hc128       ... bench:     445,483 ns/iter (+/- 16,674) = 2298 MB/s
test gen_u32_hc128         ... bench:       2,809 ns/iter (+/- 194) = 1423 MB/s
test gen_u64_hc128         ... bench:       4,254 ns/iter (+/- 360) = 1880 MB/s
test init_hc128            ... bench:       4,539 ns/iter (+/- 412)
test reseeding_hc128_bytes ... bench:     451,584 ns/iter (+/- 25,463) = 2267 MB/s
test reseeding_hc128_u32   ... bench:       3,690 ns/iter (+/- 125) = 1084 MB/s
test reseeding_hc128_u64   ... bench:       5,907 ns/iter (+/- 157) = 1354 MB/s
```
And before this PR:
```
test reseeding_hc128_bytes ... bench:     449,635 ns/iter (+/- 5,755) = 2277 MB/s
test reseeding_hc128_u32   ... bench:       6,418 ns/iter (+/- 84) = 623 MB/s
test reseeding_hc128_u64   ... bench:       7,693 ns/iter (+/- 124) = 1039 MB/s
```

This makes sense, because the overhead of checking and indexing in the results array makes up 20~40%. That part is now doubled because it also does checks for reseeding.

You are right, this benchmark is testing something nonsensical. Reseeding Xorshift ?!. I don't think it matters much because they both show the overhead of `ReseedingRng`, but I'll change it.